pym/bob/cmds/build.py

```diff
@@ -976,15 +1010,36 @@ def _cookPackageStep(self, packageStep, checkoutOnly, depth):
                     self._runShell(packageStep, "package")
                     packageHash = hashWorkspace(packageStep)
                     audit = self._generateAudit(packageStep, depth, packageHash)
+                    # copy to shared location if possible
+                    if self.__installShared and packageStep.isShared() and packageStep.isPackageRelocatable() and self.__sharedFolder:
+                        buildidhexstr =  asHexStr(packageBuildId)
+                        sharedpackagepath = os.path.join(self.__sharedFolder, buildidhexstr)
+                        print(colorize("     Install in shared location {}".format(sharedpackagepath), "32"))
+                        # create temp dir
+                        tmpdir = tempfile.mkdtemp()
```

Comment:
You have to create the temporary directory in the shared folder. Otherwise the rename is not possible if /tmp is on another file system. At least the rename cannot be atomic anymore.